api/client/src/main/java/jakarta/websocket/ClientEndpoint.java

```diff
@@ -78,4 +78,16 @@
      *         annotation.
      */
     public Class<? extends ClientEndpointConfig.Configurator> configurator() default ClientEndpointConfig.Configurator.class;
+
+    Extension[] extensions() default {};
+
+    @interface Extension {
+        String name();
+        Parameter[] parameters() default {};
+    }
+
+    @interface Parameter {
+        String name();
+        String value();
```

Comment:
Some parameters do not require a value such as `server_no_context_takeover` used in `permessage-deflate`, so perhaps this should have a default.

Reply:
OK. I'll include that in the next round of updates.